src/main/java/org/truffleruby/core/cast/DurationToMillisecondsNode.java

```diff
@@ -65,12 +67,25 @@ protected long durationNil(Nil duration) {
     }
 
     @Specialization
-    protected long duration(RubyDynamicObject duration) {
+    protected long duration(RubyDynamicObject duration,
+            @Cached DispatchNode divmodNode,
+            @Cached DispatchNode sumNode,
+            @Cached ToLongNode toLongNode) {
         if (floatCastNode == null) {
             CompilerDirectives.transferToInterpreterAndInvalidate();
             floatCastNode = insert(NumericToFloatNodeGen.create());
         }
-        return duration(floatCastNode.executeDouble(duration));
+        try {
+            return duration(floatCastNode.executeDouble(duration));
+        } catch (Exception e) {
+            try {
+                Object result = divmodNode.call(duration, "divmod", 1);
```

Comment:
In CRuby this is `rb_check_funcall(num, id_divmod, 1, &i)`, so it does not raise if the method does not exist, and instead throws a TypeError if it the method cannot be found.

We can do the same with DispatchNode.PRIVATE_RETURN_MISSING which returns `DispatchNode.MISSING` if not found (there are a few example usages in the codebase).